File: SEM-3/Java-OOP/lab7/Maximum.java
package mypackages.p1;
public class Maximum{
    public static int maximum(int a, int b, int c){
        if(a>b && a>c) return a;
        else if(b>a && b> c) return b;
        else return c;
    }
    public static float maximum(float a, float b , float c){
        if(a>b && a>c) return a;
        else if(b>a && b> c) return b;
        else return c;
    }
    public static int maximum(int [] arr){
        int l = arr[0];
        for(int x : arr){
            if(x>l) l=x;
        }
        return l;
    }
    public static int maximum(int [][] arr){
        int l = arr[0][0];
        for(int one[] : arr){
            for(int x : one){
                if(x>l) l=x;
            }
        }
        return l;
    }
}

File: PSUC/lab10-ModularProgramming/maximum.c
//to find the maximum number in the input integer array using pointers
#include<stdio.h>

int main(){
  int n, arr[20];
  printf("Enter array size: ");
  scanf("%d",&n);
  printf("Enter array elements: ");
  for(int i = 0 ; i < n ; i++){
    scanf("%d", &arr[i]);
  }
  int *f = arr;
  int *l = arr;
  for(int i = 1; i<n ; i++){
    if(*(l+i) > *f){
      f = l + i;
    }
  }
  printf("Maximum element: %d", *f);
}